tests/cli/test_nb.py

```diff
@@ -24,6 +24,217 @@ def git_commit():
+    nb_inject_assets_path = f'{path_to_assets}/test-nb-inject-assets'
+    template_path = f'{nb_inject_assets_path}/template.ipynb'
+    template_b_path = f'{nb_inject_assets_path}/template_b.ipynb'
+    test_pipeline = f'{nb_inject_assets_path}/pipeline.yaml'
+
+    Path(test_pipeline).write_text(f"""
+tasks:
+  - source: {template_path}
+    name: task-a
+    product: {nb_inject_assets_path}/report-a.ipynb
+    params:
+      some_param: a
+
+  - source: {template_b_path}
+    name: task-c
+    product: {nb_inject_assets_path}/report-c.ipynb
+    params:
+      some_param: c
+    """)
+
+    inject_results = []
+    param_to_inject_a = 'a'
+    param_to_inject_b = 'c'
+
+    monkeypatch.setattr(sys, 'argv', [
+        'ploomber', 'nb', '--entry-point', test_pipeline,
+        '--inject', '--priority',
+        f'task-{param_to_inject_a}', '--priority', f'task-{param_to_inject_b}'
+    ])
+
+    cli.cmd_router()
+    out, err = capsys.readouterr()
+    if err:
+        inject_results.append(False)
+    else:
+        injected_params_a = get_nb_injected_params(template_path)
+        inject_results.append(injected_params_a == f'"{param_to_inject_a}"')
+
+        injected_params_b = get_nb_injected_params(template_b_path)
+        inject_results.append(injected_params_b == f'"{param_to_inject_b}"')
+
+    assert all(inject_results)
+
+
+def test_inject_multiple_task_parameters_that_use_the_same_template(
+        monkeypatch, capsys, path_to_assets):
+    # ploomber nb --inject --priority task-a --priority task-b
+    nb_inject_assets_path = f'{path_to_assets}/test-nb-inject-assets'
+    template_path = f'{nb_inject_assets_path}/template.ipynb'
+    test_pipeline = f'{nb_inject_assets_path}/pipeline.yaml'
+
+    Path(test_pipeline).write_text(f"""
+tasks:
+  - source: {template_path}
+    name: task-a
+    product: {nb_inject_assets_path}/report-a.ipynb
+    params:
+      some_param: a
+
+  - source: {template_path}
+    name: task-b
+    product: {nb_inject_assets_path}/report-b.ipynb
+    params:
+      some_param: b
+    """)
+
+    param_to_inject_a = 'a'
+    param_to_inject_b = 'b'
+
+    monkeypatch.setattr(sys, 'argv', [
+        'ploomber', 'nb', '--entry-point', test_pipeline,
+        '--inject', '--priority',
+        f'task-{param_to_inject_a}', '--priority', f'task-{param_to_inject_b}'
+    ])
+
+    with pytest.raises(BaseException):
+        cli.cmd_router()
+
+    out, err = capsys.readouterr()
+    assert 'Values are correspond to the same task' in err
+
+
+def test_inject_invalid_prioritized_task_single_task(monkeypatch, capsys,
+                                                     path_to_assets):
+    # ploomber nb --inject --priority this-task-doesnt-exist
+    nb_inject_assets_path = f'{path_to_assets}/test-nb-inject-assets'
+    template_path = f'{nb_inject_assets_path}/template.ipynb'
+    test_pipeline = f'{nb_inject_assets_path}/pipeline.yaml'
+
+    Path(test_pipeline).write_text(f"""
+tasks:
```

Comment:
This and the 3 lines below:

```
nb_inject_assets_path = f'{path_to_assets}/test-nb-inject-assets'
template_path = f'{nb_inject_assets_path}/template.ipynb'
test_pipeline = f'{nb_inject_assets_path}/pipeline.yaml'
```

Are repetitive, let's export it to a function and reuse the code.